h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -489,44 +505,82 @@ public MP nextModelParameters(Model previousModel) {
           // NOTE: nextModel checks _visitedHyperparamIndices and does not return a duplicate set of indices.
           // NOTE: in RandomDiscreteValueWalker nextModelIndices() returns a new array each time, rather than
           // mutating the last one.
+
           _currentHyperparamIndices = nextModelIndices();
 
           if (_currentHyperparamIndices != null) {
-            _visitedPermutations.add(_currentHyperparamIndices);
-            _visitedPermutationHashes.add(integerHash(_currentHyperparamIndices));
-            _currentPermutationNum++; // NOTE: 1-based counting
+              _visitedPermutations.add(_currentHyperparamIndices);
```

Comment:
@honzasterba sure, this is obviously incidental because it is not aligned properly right now. So I will move it back and cause same issue once again :)